app/exec/build/tasks/sign.ts

```diff
@@ -0,0 +1,159 @@
+import admZip = require("adm-zip");
+import archiver = require("archiver");
+import del = require('del');
+import fs = require("fs");
+var ncp = require('ncp').ncp;
+import os = require('os');
+import path = require("path");
+import shell = require("shelljs");
+import tasksBase = require("./default");
+import trace = require("../../../lib/trace");
+
+export interface TaskSignResult {
+	signingSuccessful: boolean;
+}
+
+export function getCommand(args: string[]): BuildTaskSign {
+	return new BuildTaskSign(args);
+}
+
+export class BuildTaskSign extends tasksBase.BuildTaskBase<TaskSignResult> {
+	protected description = "Sign a task.";
+
+	constructor(args: string[]) {
+		super(args);
+	}
+
+	protected getHelpArgs(): string[] {
+		return ["certFingerprint", "taskPath", "newGuid", "newNameSuffix"];
+	}
+
+	// tfx build tasks sign --task-path {TASK_PATH} --cert-fingerprint {FINGERPRINT} --new-guid {GUID} --new-name-suffix ' - SIGNED'
+	// --new-guid and --new-name-suffix are optional
+	public async exec(): Promise<TaskSignResult> {
+		const taskZipsPath: string[] | null = await this.commandArgs.taskPath.val();
+		const taskZipPath = taskZipsPath[0];
+		const certFingerprint: string | null = await this.commandArgs.certFingerprint.val();
+		const newGuid: string | null = await this.commandArgs.newGuid.val();
```

Comment:
Since `newGuid` and `newNameSuffix` are optional, I believe you want to pass `true` into `val()` so that it does not prompt the user for the value.

Reply:
I think setting `null` as the default value does that, right?

Reply:
yep, you're right.